tests/e2e/specs/shopper/checkout-different-addresses.test.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import {
+	merchant,
+	openDocumentSettingsSidebar,
+	uiUnblocked,
+} from '@woocommerce/e2e-utils';
+
+import {
+	visitBlockPage,
+	selectBlockByName,
+	saveOrPublish,
+} from '@woocommerce/blocks-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	shopper,
+	preventCompatibilityNotice,
+	reactivateCompatibilityNotice,
+} from '../../../utils';
+
+import { BILLING_DETAILS, SHIPPING_DETAILS } from '../../../utils/constants';
+const SIMPLE_PRODUCT_NAME = '128GB USB Stick';
+
+const config = require( 'config' );
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( 'Skipping Checkout tests', () => {} );
+
+describe( 'Shopper → Checkout → Can have different shipping and billing addresses', () => {
+	beforeAll( async () => {
+		// prevent CartCheckoutCompatibilityNotice from appearing
+		await preventCompatibilityNotice();
+		await merchant.login();
+
+		// Activate all address fields.
+		await visitBlockPage( 'Checkout Block' );
+		await openDocumentSettingsSidebar();
+		await selectBlockByName(
+			'woocommerce/checkout-shipping-address-block'
+		);
+		await expect( page ).toClick( 'label', { text: 'Company' } );
+		await expect( page ).toClick( 'label', {
+			text: 'Apartment, suite, etc.',
+		} );
+		await expect( page ).toClick( 'label', { text: 'Phone' } );
```

Comment:
If I understand correctly, we want to activate all shipping address fields by toggling on the checkboxes in the image below:

<img width="244" alt="image" src="https://user-images.githubusercontent.com/14235870/157704393-f5a8ec1c-fe15-4802-a2da-07aa359423da.png">

And to do that, we are looking for the corresponding label by name then clicking on it. There are two issues with this approach:

- We aren't checking if a field is already active. Initially, only the "Company" field is disabled. The other two fields (i.e., phone & apartment) are already active.
- The **Shipping address** form also contains a label with the same name. (See image below)
<img width="320" alt="image" src="https://user-images.githubusercontent.com/14235870/157705823-bd6add2d-eb6e-4242-bef2-869438fc4d0a.png">
<img width="287" alt="image" src="https://user-images.githubusercontent.com/14235870/157705893-9a825ba1-d4ef-4b17-acca-eea6a1c5dc89.png">
<img width="242" alt="image" src="https://user-images.githubusercontent.com/14235870/157705943-20df05d9-8210-469b-b5e6-870d8ecc7827.png">

This means, that if a field is already active/displayed we are actually clicking on the form instead of the checkboxes.

Since the **"Phone"** and **"Apartment"** fields are active by default, I suggest we only target the **"Company"** field. And, instead of looking for the label, we can use the `setCheckbox` utility function from Woo:

```js
/**
 * External dependencies
 */
import { setCheckbox } from '@woocommerce/e2e-utils';

// Activate the "Company" field
await setCheckbox('#inspector-toggle-control-1');
```

Reply:
Changed! 😀